sklearn/metrics/tests/test_classification.py

```diff
@@ -1348,26 +1352,34 @@ def test_precision_recall_f1_score_with_an_empty_prediction():
 
 @pytest.mark.parametrize('beta', [1])
 @pytest.mark.parametrize('average', ["macro", "micro", "weighted", "samples"])
-def test_precision_recall_f1_no_labels(beta, average):
+@pytest.mark.parametrize('zero_division', ["warn", 0, 1])
+def test_precision_recall_f1_no_labels(beta, average, zero_division):
     y_true = np.zeros((20, 3))
     y_pred = np.zeros_like(y_true)
 
-    p, r, f, s = assert_warns(UndefinedMetricWarning,
-                              precision_recall_fscore_support,
-                              y_true, y_pred, average=average,
-                              beta=beta)
-    assert_almost_equal(p, 0)
-    assert_almost_equal(r, 0)
-    assert_almost_equal(f, 0)
+    func = precision_recall_fscore_support
+    my_assert = (assert_warns if zero_division == "warn"
+                 else assert_no_warnings)
+    tmp = ([UndefinedMetricWarning, func] if zero_division == "warn"
+           else [func])
+    p, r, f, s = my_assert(*tmp, y_true, y_pred, average=average,
+                           beta=beta, zero_division=zero_division)
+    tmp = ([UndefinedMetricWarning, fbeta_score] if zero_division == "warn"
+           else [fbeta_score])
+    fbeta = my_assert(*tmp, y_true, y_pred, beta=beta,
+                      average=average, zero_division=zero_division)
+
+    zero_division = float(zero_division == 1)
```

Comment:
obfuscated

Reply:
simplified in the new version with two separated tests